frontend/app/src/components/dashboard/OverallBalances.vue

```diff
@@ -23,9 +23,20 @@ const { totalNetWorth } = storeToRefs(statistics);
 const frontendStore = useFrontendSettingsStore();
 const { visibleTimeframes } = storeToRefs(frontendStore);
 
-const { isLoading: isSectionLoading } = useStatusStore();
+const { isLoading: isSectionLoading, setStatus } = useStatusStore();
 
-const isLoading = isSectionLoading(Section.BLOCKCHAIN);
+const isBlockchainLoading = isSectionLoading(Section.BLOCKCHAIN);
+const isDashboardLoading = isSectionLoading(Section.DASHBOARD);
+const isLoading = logicOr(isBlockchainLoading, isDashboardLoading);
+
+watch(isBlockchainLoading, (loading) => {
+  if (!loading) {
+    setStatus({
+      section: Section.DASHBOARD,
+      status: Status.LOADED,
+    });
+  }
+});
```

Comment:
What happens if you navigate away before loading completes, wait say in balances for the loading to complete and then navigate back to the dashboard?

Reply:
yeah you're right, let me update

Reply:
updated, should be more straightforward now